### ISSUE TITLE
Token may contain a \t character between two numbers in certain situations

### ISSUE THREAD

**Issue by gpetho** [nytud/quntoken#45]
I'm working with an OCR'ed document that contains several tables with lots of numbers in them. When I run `emtsv tok` on it, the output contains `\t` characters in some tokens, breaking xtsv functionality, like in the following examples:

```
form wsafter
...
^	""
000	2570	"\t\t\r\n"
C.	" "
...
,	""
69	171	""
,^^	"\t"
...
3 275	"\t"
6 744 350	2100	"\t"
02/10/02	"\t"
...
-1,83	"\t"
107	105	108	106	"\t"
13	"\t"
...
Magyarország	"\t"
93	110	11900	"\t"
52	1998	"\t"
26	"\t"
...
KW	"\t"
47	200Ö	"\t"
27	"\t"
```

I have included the directly preceding and following tokens surrounding the broken tokens in the examples above. I don't see any pattern when a tab between two numbers is correctly handled as `wsafter` and when it becomes part of the token's form. The only thing that's systematic is that the tab always appears between two Arabic figures.

I'm attaching a small raw text file that yields lots of broken tokens when it's processed with emtsv tok.
[example_bad_tabs.txt](https://github.com/nytud/quntoken/files/7883538/example_bad_tabs.txt)

I have reproduced the issue on two different machines.

**Comment by sassbalint** [comment]
We can say that no `\t` characters are allowed in the input text.
Do we have a better solution than this currently?

**Comment by gpetho** [comment]
> Do we have a better solution than this currently?

My workaround for the original issue was to filter out all invalid lines before passing the output of tok to the downstream modules basically like this: `cat example_bad_tabs.txt | emtsv tok | awk 'NF < 3' | emtsv morph`
I would personally prefer this to excluding all tab characters from the raw input. That would seem like a bit of an overkill.

**Comment by sassbalint** [comment]
> > Do we have a better solution than this currently?
> 
> My workaround for the original issue was to filter out all invalid lines before passing the output of tok to the downstream modules basically like this: `cat example_bad_tabs.txt | emtsv tok | awk 'NF < 3' | emtsv morph` I would personally prefer this to excluding all tab characters from the raw input. That would seem like a bit of an overkill.

Yes, kind of, but using your solution we lose some tokens from the original text.

**Comment by gpetho** [comment]
> Yes, kind of, but using your solution we lose some tokens from the original text.

Right, the question is how "no `\t` characters are allowed in the input text" should be implemented. If all lines containing tabs are filtered out before running the tokenizer, then many more tokens are lost. If all tabs are replaced by spaces before running the tokenizer, then the distinction between tabs and spaces is lost. The filter I'm proposing is a compromise that tries to avoid the full impact of both of these alternatives.

**Comment by gpetho** [comment]
How about this then?
`cat example_bad_tabs.txt | emtsv tok | awk 'NF < 3 {print $0}; NF >=3 { for (i = 1; i < NF-1; i++) {printf("%s\t\"\\t\"\n", $i)}; printf("%s\t%s\n", $(NF-1), $NF) }' | emtsv morph`

This does what the tokenizer would be normally expected to do: split the malformed `form`s by the tabs, put the tab characters in `wsafter`, and leave the last segment of the `form` and its `wsafter` alone.
